langchain/src/vectorstores/singlestore.ts

```diff
@@ -4,18 +4,76 @@ import type {
   OkPacket,
   ResultSetHeader,
   FieldPacket,
+  PoolOptions,
 } from "mysql2/promise";
 import { format } from "mysql2";
+import { createPool } from "mysql2/promise";
+import { Metadata } from "@opensearch-project/opensearch/api/types.js";
```

Comment:
Oh, you should add this to the docs as a peer dependency and use `import type`.

Reply:
I prefer to redeclare this type inside my code.